keps/sig-release/2572-release-cadence/README.md

```diff
@@ -0,0 +1,311 @@
+
+## Proposal
+
+### User Stories
+
+Kubernetes releases are made by real people. The technical aspects—for example,
+the release automation—reflects only a tiny part of the complete cycle. This
+means we will mainly focus on the human aspects and their corresponding roles
+when deciding to move to a 3-releases-per-year cadence.
+
+#### End User
+
+Most end user organizations find it difficult to match Kubernetes release
+cadence - only 3 releases per year will relax this situation.
+
+#### Distributors and downstream projects
+
+Downstream projects assemble their solution from different projects. Having
+fewer upgrades helps them to reduce complexity. For example, cloud providers
+will gain more room for upgrading their infrastructure.
+
+#### Contributors
+
+With a lower release cadence, contributors will gain more time for project
+enhancements, feature development, planning, and testing. It will provide more
+room for maintaining their mental health, prepare for events like KubeCon or
+work on the downstream integration.
+
+Through this proposal SIG Release's aim is to give contributors more
+flexibility to decide how to invest their time. It is explicitly _not_ to push
+contributors in doing more.
+
+#### SIG Release members
+
+By applying a cadence of 3 releases per year, SIG Release members will gain a
+reduced management overhead. There are also only 3 patch releases to maintain,
+which right now can overlap up to 4. SIG Release will gain more time to ensure
+a seamless transition from the previous release team to the next one. It is
+also possible to include more shadows if the role leads conclude that this is
+appropriate.
+
+### Risks and Mitigations
+
+#### Concentrating risk
+
+In theory a reduced release cadence will cause more changes for every release.
+This means that there will be an increased risk, which would usually be split
+up into 4 dedicated milestones rather than 3.
+
+SIG Release cannot mitigate this risk directly, but is able to track and
+influence it during each release cycle. It's the responsibility of SIG Release,
+together with SIG Testing and SIG Architecture, to identify new gaps and issues
+in the release cadence and mitigate them on a case-by-case basis.
+
+#### Attention to tests
+
+This KEP does not propose any change to the release cycle itself and assumes
+that the same periods for Code and Test Freeze. Assuming that, there is an
+increased risk for flakes and test failures. It will be the responsibility of
+SIG Release to mitigate this, together with the CI signal role. If we speak
+about an overall release cycle enhancement of 3-4 weeks, then we believe that
+SIG Release is able to mitigate this risk over multiple releases.
+
+#### Attention to dependencies
+
+Having fewer releases will introduce the risk of missing dependencies — for
+example, Golang upgrades. This has to be mitigated on a case-by-case basis, in
+the same way as it is being done right now.
```

Comment:
Add:

#### Slowed Feature Advancement

Research discovered that [only 5% of Kubernetes features](https://github.com/kubernetes/sig-release/discussions/1290#discussioncomment-336781) advanced from Alpha to GA in the minimum 3 releases.  However, the same research showed that reminders from the Release Team played a critical role in advancement of more than 50% of features.   With a 4-month release cycle, this reminder activity can be expected to slow down.  As such, advancement will need to be mitigated by somehow making sure that SIGs/authors remember to work on feature advancement by some other means.

Reply:
Been working at this and would welcome a conversation and support :)

Reply:
ah, we've been discussing it in sig-release separately.  should coordinate!

Reply:
ALso ... please add text to the above if you have a plan!

Reply:
Done, added a dedicated feature graduation section. :heavy_check_mark:

Reply:
The ideas described here, created and adopted by SIGs across the project, can be helpful: https://github.com/kubernetes/community/blob/37944bbefe93e77b635706c7357f4da37c0d3d3f/contributors/chairs-and-techleads/process-workflow-guide.md. The sections on KEP planning and roadmapping are based on some standard practices that enable teams to be proactive and transparent about their goal-setting in advance of deadlines. It would be great to see broader adoption of these general practices to help turn successful one-off approaches into norms. If we'd like to explore ways to encourage that adoption more systematically across the project, let's have that conversation—I'm confident it can be done in a way that preserves SIG autonomy / isn't top-down, onerous, or cookie-cutter.

At a personal level, I've recently been working with a few groups beyond SIG Release to establish some of the practices described in the process guide. These efforts are geared toward helping SIGs to get ahead of their workloads and plans, work more sustainably, be more consistently responsive to contributors and other SIGs, and mitigate risks. Groups that have requested and accepted the help include SIG Windows and the Structured Logging WG. More are in the immediate pipeline, and momentum appears to be building. My favoured approach is to start with an actual, specific delivery goal—setting up a new WG, breaking down a complex deliverable into something achievable within a given timeframe, organising splintered and scattered backlogs into coherent + prioritized roadmaps—and baking process change into that effort so that it's not distracting. 

Another idea being piloted right now is the Enhancements Liaison role. This idea is still crystallising, and is an effort to address some of the concerns about SIG responsiveness. In this role, time-boxed to a release cycle, a contributor embeds with a specific SIG (ideally one they are willing to stay with long-term) and commits during the cycle to support chairs in coordinating and communicating with the release team. These contributors will be onboarded and supported during 1.22. A few SIGs (Autoscaling, Windows) have expressed interest in working with a liaison and a few others are considering it. I understand that an untested idea doesn't make for a compelling argument, but it's a start.